shared/src/main/scala/mlscript/JSBackend.scala

```diff
@@ -556,64 +556,36 @@ class JSBackend(allowUnresolvedSymbols: Boolean) {
 
   private def translateParents(superFields: Ls[Term], constructorScope: Scope)(implicit scope: Scope): Opt[JSExpr] = {
     val bases = superFields.map { sym => sym match {
-      case App(lhs, _) => S(translateTerm(App(lhs, Tup(Ls())))(constructorScope))
-      case _ => S(translateTerm(sym)(constructorScope))
+      case App(lhs, _) => translateTerm(App(lhs, Tup(Ls())))(constructorScope)
+      case _ => translateTerm(sym)(constructorScope)
     } }
 
+    def translateParent(current: JSExpr, base: JSExpr, forceMixin: Bool): JSExpr = {
+      val (name, captured) = current match {
+        case JSIdent(nme) => (nme, false)
+        case JSInvoke(JSIdent(nme), _) => (nme, false)
+        case f: JSField => (f.property.name, true)
+        case JSInvoke(f: JSField, _) => (f.property.name, true)
+        case _ => throw CodeGenError("unsupported parents.")
+      }
+
+      scope.resolveValue(name) match {
+        case Some(CapturedSymbol(out, sym: MixinSymbol)) if (captured) =>
+          JSInvoke(translateCapture(CapturedSymbol(out, sym)), Ls(base))
+        case Some(CapturedSymbol(out, sym: NuTypeSymbol)) if (captured && !forceMixin) =>
+          translateCapture(CapturedSymbol(out, sym)).member("class")
+        case Some(sym: MixinSymbol) if (!captured) =>
+          JSInvoke(translateVar(name, false), Ls(base))
+        case Some(_: NuTypeSymbol) if (!captured && !forceMixin) =>
+          translateVar(name, false).member("class")
+        case _ => throw CodeGenError(s"unresolved parent $name.")
```

Comment:
It seems the code above is implicitly assuming that if the parent is a JSField (possibly invoked) then we'll get a `CapturedSymbol`. Why do you do that? What's the point of the local `capture` flag?

Reply:
It seems that the local variable is redundant and it has been removed